src/mudlet-lua/lua/LuaGlobal.lua

```diff
@@ -97,7 +97,25 @@ function unzip( what, dest )
   z:close()
 end
 
+--[[
+Effectively this has migrated from cTelnet::handle_socket_signal_connected()
+but can be replaced with something more advanced by the player if needed. It
+would also be possible to produce a sysConnectionEvent handler or a simple
+pair) of triggers. In either of those two cases it would be necessary to replace
+this predefined function in the profile's scripts with a dummy, do nothing one,
+of the form:
+
+  function doLogin()
+  end
 
+so that the Lua interpreter doesn't run this code.
+]]
+function doLogin()
+  if getPlayerName() ~= "" then
```

Comment:
Let's make `getPlayerName()` return `nil` if there isn't one, so we can just do `if getPlayerName() then` which is more idiomatic Lua.

Reply:
But it *isn't* a run-time error that justifies testing for emptiness and putting up both a `nil` and an error message is it? :man_shrugging:

Reply:
Not an error, but it is a lack of a value, which is perfect for `nil`. Don't need the `msg` part.

Reply:
But, *but*,  **but** - I was under the (de)illusion that a `nil` return *is* the way that Lua says - "*Nope! You can't do that with this, at least, not right now...*" but it **must** be accompanied with a message saying something along the lines: "you cannot bar() without a foo". :confounded:

Reply:
It's a good pattern in case of an error, but this isn't an error, just no value. So it's ok!

Reply:
Compromise time - I'll return the `nil` that you others wanted; and I will tack on:
`no character name set, so nothing to return` to match the style of the error messages from the other two - and so my modified `run-lua-code.xml` which DOES report any messages associated with a primary `nil` return value has something to show...

in next commit

Reply:
But it's not an error to have no value in that field 🤔 do you think it's a mandatory value?